pkg/util/errors.go

```diff
@@ -282,6 +285,24 @@ func HandleInstanceSharingError(err error, entityName string) error {
 			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, does not match the service offering.", entityName),
 			StatusCode:  http.StatusNotFound,
 		}
+	case ErrMultipleReferenceSelectorResults:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: "Failed to create the reference. There are multiple shared instances that match your criteria. Specify explicitly the instance id you would like to refer to.",
```

Comment:
"Failed to create the reference. There are multiple shared instances that match your criteria. Specify the ID of the instance to which you want to refer."